pkgs/development/python-modules/geocat-viz/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib
+, fetchPypi
+, buildPythonPackage
+, pythonOlder
+, setuptools
+, numpy
+, xarray
+, matplotlib
+, cartopy
+, pip
+, setuptools-scm
+}:
+
+let
+  pname = "geocat.viz";
+  version = "2023.10.0";
+in
+buildPythonPackage {
+  inherit pname version;
+  format = "pyproject";
+  disabled = pythonOlder "3.9";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-IX/c82KGGsNr25FNKF7EqI8GYzv/b45Pqn9csqnfYtU=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    numpy
+    xarray
+    matplotlib
+    cartopy
+    pip
+    setuptools-scm
+  ];
```

Comment:
Upstream mention some of those inputs as [runtime requirement](https://github.com/NCAR/geocat-viz/blob/main/setup.cfg#L38C1-L46C10). They goes into `propagatedBuildInputs`. `nativeBuildInputs` should cover `setup_requires`.